src/convertToObject.js

```diff
@@ -6,7 +6,27 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const result = {};
+  const splited = sourceString.split(';');
+
+  for (const item of splited) {
+    if (item.trim() === '') {
+      continue;
+    }
+
+    const [prop, value] = item.split(':');
+
+    if (!prop || !value) {
+      continue;
+    }
+
+    const trimmedProp = prop.trim();
+    const trimmedValue = value.trim();
+
+    result[trimmedProp] = trimmedValue;
```

Comment:
Instead of using a loop, consider utilizing array iteration methods like 'map' and 'filter' to transform the string into an object as per the task instructions. This would result in cleaner and more functional-style code.